File: ios/Runner/Repository/Contract/ArtistRepositoryContract.swift
//
//  ArtistRepositoryContract.swift
//  Runner
//
//  Created by t k on 2024/08/12.
//

protocol ArtistRepositoryContract {
    func findAll() -> [ArtistModelContract]

    func findById(artistId: ArtistId) -> ArtistModelContract?
    
    func findByIds(artistIds: [ArtistId]) -> [ArtistModelContract]
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/repository/contract/ArtistRepositoryContract.kt
package dev.tcode.thinmpf.repository.contract

import dev.tcode.thinmpf.model.contract.ArtistModelContract
import dev.tcode.thinmpf.model.valueObject.ArtistId

interface ArtistRepositoryContract {
    fun findAll(): List<ArtistModelContract>

    fun findById(artistId: ArtistId): ArtistModelContract?

    fun findByIds(artistIds: List<ArtistId>): List<ArtistModelContract>
}